src/tennissimulator/utils/IdGenerator.java

```diff
@@ -0,0 +1,11 @@
+package tennissimulator.utils;
+
+public class IdGenerator {
+
+    public static int playersCounter = 0;
```

Comment:
Since this class and this variable is about ids, I would suggest naming it "id". Imagine you will have different ID generation strategy in the future, such change shouldn't need changing name of the variable in my opinion.

And it can be private

Reply:
fixed